userapi/src/main/java/fk/prof/userapi/model/tree/IndexedTreeNode.java

```diff
@@ -0,0 +1,71 @@
+package fk.prof.userapi.model.tree;
+
+import fk.prof.userapi.model.Tree;
+
+import java.util.List;
+
+/**
+ * Represents a indexable node in the tree. This node has an index, accompanying data and the list of children nodes.
+ * Created by gaurav.ashok on 01/06/17.
+ */
+public class IndexedTreeNode<T> {
+    public IndexedTreeNode(int idx, T data) {
+        this(idx, data, null);
+    }
+
+    public IndexedTreeNode(int idx, T data, List<IndexedTreeNode<T>> children) {
+        this.idx = idx;
+        this.data = data;
+        this.children = children;
+    }
+
+    private final int idx;
+
+    private final T data;
+
+    private List<IndexedTreeNode<T>> children;
+
+    public int getIdx() {
+        return idx;
+    }
+
+    public T getData() {
+        return data;
+    }
+
+    public List<IndexedTreeNode<T>> getChildren() {
```

Comment:
Can usages of this be replaced with visit method? I remember JJ's strong reservations around exposing getters for mutable members.
Usage in unit tests can be done by making this  package visible

Reply:
Made it package visible